python/ray/air/utils/data_batch_conversion_utils.py

```diff
@@ -0,0 +1,90 @@
+from typing import Type
+
+import numpy as np
+import pandas as pd
+
+from ray.air.data_batch_type import DataBatchType
+from ray.air.constants import TENSOR_COLUMN_NAME
+from ray.util.annotations import DeveloperAPI
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pass
+
+
+@DeveloperAPI
+def convert_batch_type_to_pandas(data: DataBatchType) -> pd.DataFrame:
+    """Convert the provided data to a Pandas DataFrame.
+
+    Args:
+        data: Data of type DataBatchType
+
+    Returns:
+        A pandas Dataframe representation of the input data.
+
+    """
+    from ray.air.utils.tensor_extensions.pandas import TensorArray
+
+    if isinstance(data, pd.DataFrame):
+        return data
+
+    elif isinstance(data, np.ndarray):
+        if data.dtype == object:
+            return pd.DataFrame({TENSOR_COLUMN_NAME: list(data)})
```

Comment:
Do we want this or do we want `data.tolist()`? https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html

Reply:
@krfricke I think that our assumption here is that `dtype == object` => this is an ndarray containing a collection of heterogeneous ndarrays (heterogeneous types, shapes, etc.), so we only want to create a top-level list of ndarrays here.

Actually, @amogkam, is this list conversion required? `TensorArray` already has dedicated logic for handling opaque object type ndarrays, and the list path seems similar. We should make sure this is properly covered by tests.

Reply:
Yep that's right, but using `TensorArray` means the resulting dataframe does not have full feature parity with a regular Pandas dataframe.

For example, a `df.info` will fail. I'd prefer not using `TensorArray` unless neede.

Reply:
Ah that's right, thanks!